sdks/unreal/Agones/Source/Agones/Private/AgonesComponent.cpp

```diff
@@ -83,6 +99,36 @@ void UAgonesComponent::Connect()
 	ConnectDel.BindUObject(this, &UAgonesComponent::GameServer, SuccessDel, FAgonesErrorDelegate());
 	GetWorld()->GetTimerManager().ClearTimer(ConnectDelTimerHandle);
 	GetWorld()->GetTimerManager().SetTimer(ConnectDelTimerHandle, ConnectDel, 5.f, true);
+
+    if (!bWatchGameServer || !FModuleManager::LoadModulePtr<FWebSocketsModule>(TEXT("WebSockets")))
+    {
+        return;
+	}
+
+    if (!WatchWebSocket)
+    {
+        TMap<FString, FString> Headers;
+
+        // Make up a WebSocket-Key value. It can be anything!
+        Headers.Add(TEXT("Sec-WebSocket-Key"), FGuid::NewGuid().ToString(EGuidFormats::Short));
+        Headers.Add(TEXT("Sec-WebSocket-Version"), TEXT("13"));
+        Headers.Add(TEXT("User-Agent"), TEXT("X-UnrealEngine-Agent"));
+
+        // Unreal WebSockets are not able to do DNS resolution for localhost for some reason
+        // so this is using the IPv4 Loopback Address instead.
+        WatchWebSocket = FWebSocketsModule::Get().CreateWebSocket(FString::Format(TEXT("ws://127.0.0.1:{0}/watch/gameserver"), {*HttpPort}), TEXT(""));
+    }
+    else
+    {
+        WatchWebSocket->OnRawMessage().Clear();
+    }

```

Comment:
When do we go into this part of the conditional?
Also is `WatchWebSocket` not `nullptr` at this point?